src/auth-service/utils/create-preference.js

```
@@ -54,35 +54,66 @@
+      update["$set"][field] = uniqueArray;
       delete update[field];
     }
   });
 
-  // Remove simple ObjectId fields from being processed by $addToSet
+  // Process single ObjectId fields
   const singleObjectIdFields = ["user_id", "group_id"];
   singleObjectIdFields.forEach((field) => {
     if (update[field]) {
+      // Ensure single ObjectId fields are processed as-is
+      update[field] = update[field];
```

Biome (1.9.4)

[error] field is assigned to itself.

This is where is assigned.


(lint/correctness/noSelfAssign)